vendor/engines/secure_rooms/app/models/secure_rooms/user_extension.rb

```diff
@@ -8,6 +8,11 @@ module UserExtension
 
     included do
       validates :card_number, uniqueness: { allow_blank: true }
+      validates :i_class_number, uniqueness: { allow_blank: true }
+
+      def self.find_by_card_number(card_number) 
```

Comment:
This method name could be confusing because Rails [magically defines finder methods](https://guides.rubyonrails.org/active_record_querying.html#dynamic-finders) for each attribute.  This is not going to cause an issue with behavior, just future readability.  

If I saw this:
`User.find_by_card_number(params[:card_number])`

 I would assume it was doing
`User.find_by(card_number: params[:card_number])`

which is what Rails does under the hood.

```suggestion
      def self.for_card_number(card_number) 
```